### PR TITLE
autoparkLevel02

### PR DESCRIPTION
**Pull request by dzmitri-litvinov**

### CODE REVIEW

File: autopark/Vehicle.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            else
+                return 0;
+        }
+
+        public override bool Equals(object obj)
+        {
+            Vehicle temp = obj as Vehicle;
+
+            if (temp == null)
+                throw new Exception("Impossible object to compare...");
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: Don't throw exception then type are not the same. Just return false;
```suggestion
            if (temp == null)
                return fasle;
```

File: autopark/Vehicle.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+            if (VehicleType == temp.VehicleType && ModelName == temp.ModelName)
+                return true;
+            else
+                return false;
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: Unnecessary if statement. You can return bool expression result.  
```suggestion
            return VehicleType == temp.VehicleType && ModelName == temp.ModelName;
```

File: autopark/Vehicle.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        public int CompareTo(Vehicle other)
+        {
+            if (GetCalcTaxPerMonth() < other.GetCalcTaxPerMonth())
+                return -1;
+            else if (GetCalcTaxPerMonth() > other.GetCalcTaxPerMonth())
+                return 1;
+            else
+                return 0;
+        }
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: float have their own implementation of CompareTo  method. Use it to compare GetCalcTaxPerMonth return values.

File: autopark/Vehicle.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        Blue = 1,
+        White,
+        Green,
+        Gray,
+        Yellow,
+        Red
+    }
+    class Vehicle : IComparable<Vehicle>
+    {
+        public string VehicleType { get; }
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: Use class from VehicleType from Level 1 instead of string.

File: autopark/Vehicle.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                return 1.2;
+            else if (VehicleType == "types[1]")
+                return 1;
+            else if (VehicleType == "types[2]")
+                return 1.5;
+            else if (VehicleType == "types[3]")
+                return 1.2;
+            else
+                return 0;
+        }
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: Unnecessary method. Use VehileType class to implement Tax Coefficient logic.
P.S. If you write many nested ifs with equal expression for a single property use switch..case..default statment. [Click](https://metanit.com/sharp/tutorial/2.5.php)

File: autopark/Vehicle.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+{
+    enum Color
+    {
+        Blue = 1,
+        White,
+        Green,
+        Gray,
+        Yellow,
+        Red
+    }
```
Comment (UNDogMan) [inline autopark/Vehicle.cs]: Move Enum to separate file.

File: autopark/PrintVehicle.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+{
+    static class PrintVehicle
+    {
+        public static string ToString(Vehicle obj)
+        {
+            return  $"{obj.VehicleType} {obj.ModelName} " + 
+                    $"{obj.RegistrationNumber} {obj.WeightKg} " + 
+                    $"{obj.ManufactureYear} {obj.MileageKm} " + 
+                    $"{obj.Color} {obj.GetCalcTaxPerMonth().ToString("0.00")}";
+        }
```
Comment (UNDogMan) [inline autopark/PrintVehicle.cs]: Unnecessary class. Override ToString method of Vehicle class.

**Review by UNDogMan — CHANGES_REQUESTED** [review]
Check comments and do changes.

File: autopark/Vehicle.cs (hunk 1 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: Thank you. But it shows that it is impossible 
image

File: autopark/Vehicle.cs (hunk 2 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: Thank you

File: autopark/Vehicle.cs (hunk 3 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: image
Done

File: autopark/Vehicle.cs (hunk 6 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: image
Done

File: autopark/PrintVehicle.cs (hunk 7 above)
Comment (dzmitri-litvinov) [reply autopark/PrintVehicle.cs]: image
image
Done

File: autopark/Colors.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace autopark
+{
+    enum Color
```
Comment (PushMeNow) [inline autopark/Colors.cs]: Write please access modifiers for whole (enum, class, methods etc.)

File: autopark/PrintVehicle.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+            for (int i = 0; i < obj.Length; i++)
+            {
+                strBld.Append($"{obj[i].VehicleType} {obj[i].ModelName} " +
+                    $"{obj[i].RegistrationNumber} {obj[i].WeightKg} " +
+                    $"{obj[i].ManufactureYear} {obj[i].MileageKm} " +
+                    $"{obj[i].Color} {obj[i].GetCalcTaxPerMonth().ToString("0.00")}\n");
+            }
+
+            return strBld.ToString();
+        }
```
Comment (PushMeNow) [inline autopark/PrintVehicle.cs]: We have string.Join method for this action. Also  I really don't see the reason use stringbuilder here.

File: autopark/VehicleType.cs [hunk 10]
```diff
@@ -11,10 +11,10 @@ class VehicleType
         public string TypeName { get; set; }
         public float TaxCoefficient { get; set; }       
 
-        public VehicleType(string typeName, float taxCoefficient)
+        public VehicleType(string typeName)
         {
             this.TypeName = typeName;
-            this.TaxCoefficient = taxCoefficient;
+            this.TaxCoefficient = GetTaxCoefficient(typeName);
```
Comment (UNDogMan) [inline autopark/VehicleType.cs]: Return old version of VehicleType.

File: autopark/Program.cs [hunk 11]
```diff
@@ -6,33 +6,64 @@ class Program
...
+            Console.WriteLine("Vehicle with the lowest miliage:");
+            Console.WriteLine(vehicles[lowestMiliage].ToString());
+            Console.WriteLine("Vehicle with the highest miliage:");
+            Console.WriteLine(vehicles[highestMiliage].ToString());
+
+
+            Console.WriteLine("\nEqual vehicles are:");
+            bool[] isToPrint = new bool[n];
+            for (i = 0; i < n; i++)
+                isToPrint[i] = false;
```
Comment (PushMeNow) [inline autopark/Program.cs]: You know by default this array will be filled false values so you don't need this cycle.

**Review by UNDogMan — CHANGES_REQUESTED** [review]
Return old version of VehicleType

File: autopark/Vehicle.cs (hunk 5 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: Done

File: autopark/Vehicle.cs (hunk 4 above)
Comment (dzmitri-litvinov) [reply autopark/Vehicle.cs]: Done

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
I added some comments. Let me know if any questions

File: autopark/Colors.cs (hunk 8 above)
Comment (dzmitri-litvinov) [reply autopark/Colors.cs:9]: image
Done

File: autopark/Program.cs (hunk 11 above)
Comment (dzmitri-litvinov) [reply autopark/Program.cs:50]: Done

File: autopark/PrintVehicle.cs (hunk 9 above)
Comment (dzmitri-litvinov) [reply autopark/PrintVehicle.cs]: image
Done

File: autopark/Program.cs (hunk 11 above)
Comment (dzmitri-litvinov) [reply autopark/Program.cs]: Done

File: autopark/Colors.cs (hunk 8 above)
Comment (dzmitri-litvinov) [reply autopark/Colors.cs]: image
Done

File: autopark/VehicleType.cs (hunk 10 above)
Comment (dzmitri-litvinov) [reply autopark/VehicleType.cs]: image
Done

File: autopark/Program.cs [hunk 12]
```diff
@@ -6,33 +6,62 @@ class Program
...
+                        isToPrint[j] = true;
+                    }
+                }
+            }
 
-            averTaxCoef = sumTaxCoef / n;
-            Console.WriteLine($"Average tax coefficient is {averTaxCoef}");
+            for (i = 0; i < n; i++)
+                if (isToPrint[i])
+                    Console.WriteLine(vehicles[i].ToString());
```
Comment (PushMeNow) [inline autopark/Program.cs]: Just advice: don't use inline constructions because it's difficult to read and looks like ugly...

**Review by PushMeNow — COMMENTED** [review]
I left one comment. Nothing impotant for now. But in future don't use inline constructions.

File: autopark/Program.cs (hunk 12 above)
Comment (dzmitri-litvinov) [reply autopark/Program.cs]: image
Done

**Pull request merged**